client/src/app/appCtrl.js

```diff
@@ -46,36 +60,127 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
-          var contentLength = response.headers('Content-Length');
-          if (contentLength > appConfig.MAX_FILE_SIZE) {
-            slidingWindow = true;
-            $scope.view.windowing.show = true;
-          }
-          streamRemoteFile($scope.view.filePath);
-        });
-      } else {
-        downloadFile($scope.view.filePath);
-      }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+    $scope.view.file.size = getFileSize($scope.view.filePath);
+    console.log("File size "+$scope.view.file.size);
+    if ($scope.canDownload() && $scope.view.file.size > appConfig.MAX_FILE_SIZE) { 
+      console.log("streamig...");
+      downloadFile($scope.view.filePath, "streamRemote");
+    } else {
+      downloadFile($scope.view.filePath, "download"); // fallback, or small file
+    }  
+  };
+
+  // handle downloading (or Browsing) a local file
+  $scope.getLocalFile = function() {
+    // update file size for local files
+    $scope.view.file.size = getFileSize($scope.view.file.file);
+    console.log("File size "+$scope.view.file.size);
+    if ($scope.view.file.size > appConfig.MAX_FILE_SIZE) {
+      slidingWindow = true;
+      $scope.view.windowing.show = true;
+    }
+    $scope.view.loading = true;
+    downloadFile($scope.view.file.file, "streamLocal");
+  };
+
+  // main "load" function that supports both URL/local file
+  // takes care of monitoring/streaming-data plots: if appConfig.POLLING_INTERVAL > 0 keep polling the file,sleeping
+  $scope.loadFile = function(event) {
+    // react to the file-selector event
+    if (event !== null) {
+      $scope.view.file.file = event.target.files[0];
+      $scope.view.file.name = $scope.view.file.file.name;
+      $scope.view.file.path = event.target.files[0].name;
+      $scope.view.filePath = $scope.view.file.path; //TODO remove this
+    }
+    $scope.canDownload(); // will set the local file= true/false
+    loadFileHelper();
+    setMonitoringTimer(appConfig.POLLING_INTERVAL); //FIXME create an entry element for numeric value in UI for this, each change should call setMonitoringTimer()
+  };
+
+  // set up interval for cuntinuous monotoring/timer
+  // param interval: in ms, <=0 means disabled
+  // do not start parallel timers, clear existing (and optionally set new)
+  var setMonitoringTimer = function(interval) {
+    if ($scope.view.monitor.clock !== null || interval <= 0) { //disable the old one
+      clearInterval($scope.view.monitor.clock); //invalidate
+      $scope.view.monitor.clock = null;
+      $scope.view.monitor.interval = 0;
+    }
+
+    if (interval > 0) {
+      handleError("Monitoring mode started, update interval "+appConfig.POLLING_INTERVAL+"ms. ", "warning",true);
+      $scope.view.monitor.interval = interval;
+      $scope.view.monitor.clock = setInterval(function () {loadFileHelper(); console.log("updating...");}, $scope.view.monitor.interval); //FIXME work with remote too
```

Comment:
For setInterval, use Angular's $interval instead. This allows us to run automated tests on it. More information here: https://docs.angularjs.org/api/ng/service/$interval